packages/worker/src/worker.ts

```diff
@@ -993,260 +998,248 @@ export class Worker {
-
-                  return { state, output: { close, completion } };
-                } catch (err) {
-                  if (err instanceof UnexpectedError) {
-                    isFatalError = true;
-                  }
-                  throw err;
-                } finally {
-                  // Fatal error means we cannot call into this workflow again unfortunately
-                  if (!isFatalError) {
-                    // When processing workflows through runReplayHistories, Core may still send non-replay
-                    // activations on the very last Workflow Task in some cases. Though Core is technically exact
-                    // here, the fact that sinks marked with callDuringReplay = false may get called on a replay
-                    // worker is definitely a surprising behavior. For that reason, we extend the isReplaying flag in
-                    // this case to also include anything running under in a replay worker.
-                    const isReplaying = activation.isReplaying || this.isReplayWorker;
-
-                    const calls = await state.workflow.getAndResetSinkCalls();
-                    await this.processSinkCalls(calls, isReplaying);
-                  }
-                }
-              } catch (error) {
-                if (error instanceof UnexpectedError) {
-                  // rethrow and fail the worker
-                  throw error;
-                }
-                this.log.error('Failed to activate workflow', {
-                  ...(state ? state.logAttributes : { runId: activation.runId }),
-                  error,
-                  workflowExists: state !== undefined,
-                });
-                const completion = coresdk.workflow_completion.WorkflowActivationCompletion.encodeDelimited({
-                  runId: activation.runId,
-                  failed: {
-                    failure: await encodeErrorToFailure(this.options.loadedDataConverter, error),
-                  },
-                }).finish();
-                // We do not dispose of the Workflow yet, wait to be evicted from Core.
-                // This is done to simplify the Workflow lifecycle so Core is the sole driver.
-                return { state: undefined, output: { close: true, completion } };
-              }
-            },
-            undefined
-          ),
-          tap(({ close }) => {
-            this.numInFlightActivationsSubject.next(this.numInFlightActivationsSubject.value - 1);
-            if (close) {
-              group$.close();
-              this.numCachedWorkflowsSubject.next(this.numCachedWorkflowsSubject.value - 1);
-            }
-          }),
-          takeWhile(({ close }) => !close, true /* inclusive */)
-        );
+  protected handleWorkflowActivations(
+    activations$: CloseableGroupedObservable<string, coresdk.workflow_activation.WorkflowActivation>
+  ): Observable<Uint8Array> {
+    const evictions$ = this.workflowPollerStateSubject.pipe(
```

Comment:
@bergundy We still have a pending discussion regarding synthetic evictions. WDYT?

Reply:
@dandavison Supposing that we keep synthetic evictions, please rename to `syntheticEvictionActivations$` or similar. Also, add a comment above that explains that this pipe injects a synthetic eviction activation when the wf worker transition to no longer polling. I know there's a comment below, but I feel its not doing a job at helping reader understand the reason of this whole chain.

Reply:
Renamed and added comment.